x-pack/plugins/cloud/server/routes/chat.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from '../../../../../src/core/server';
+import type { SecurityPluginSetup } from '../../../security/server';
+import { GET_CHAT_USER_DATA_ROUTE_PATH } from '../../common/constants';
+import type { GetChatUserDataResponseBody } from '../../common/types';
+import { generateSignedJwt } from '../util/generate_jwt';
+
+export const registerChatRoute = ({
+  router,
+  chatIdentitySecret,
+  security,
+  isDev,
+}: {
+  router: IRouter;
+  chatIdentitySecret: string;
+  security?: SecurityPluginSetup;
+  isDev?: boolean;
+}) => {
+  router.get(
+    {
+      path: GET_CHAT_USER_DATA_ROUTE_PATH,
+      validate: {},
+    },
+    async (_context, request, response) => {
+      if (!security) {
+        return response.customError({
+          statusCode: 500,
+        });
+      }
```

Comment:
NIT/Question: What's the reasoning behind that behavior instead of just not registering the route if `security` is not present?

Reply:
That's a good idea, I'll change it.